packages/html-reporter/src/filter.ts

```diff
@@ -176,7 +176,7 @@ function cacheSearchValues(test: TestCaseSummary & { [searchValuesSymbol]?: Sear
     line: String(test.location.line),
     column: String(test.location.column),
     labels: test.tags.map(tag => tag.toLowerCase()),
-    annotations: test.annotations.map(a => a.type.toLowerCase() + '=' + a.description?.toLocaleLowerCase())
+    annotations: test.annotations.map(a => a?.type?.toLowerCase() + '=' + a.description?.toLocaleLowerCase())
```

Comment:
type is required for annotation as per https://playwright.dev/docs/api/class-testinfo#test-info-annotations. Proper fix would assert annotation types on test end.